server/monitor.go

```diff
@@ -64,8 +67,32 @@ func (s *Server) HandleConnz(w http.ResponseWriter, r *http.Request) {
 	s.mu.Lock()
 	c.NumConns = len(s.clients)
 
+	// Copy the keys to sort by them
+	pairs := make([]Pair, c.NumConns)
 	i := 0
-	for _, client := range s.clients {
+	for k, v := range s.clients {
+		pairs[i] = Pair{Key: k, Val: v}
+		i++
+	}
+
+	switch sortOpt {
+	case byCid:
+		sort.Sort(ByCid(pairs))
+	case bySubs:
+		sort.Sort(sort.Reverse(BySubs(pairs)))
+	case byOutMsgs:
+		sort.Sort(sort.Reverse(ByOutMsgs(pairs)))
+	case byInMsgs:
+		sort.Sort(sort.Reverse(ByInMsgs(pairs)))
+	case byOutBytes:
+		sort.Sort(sort.Reverse(ByOutBytes(pairs)))
+	case byInBytes:
+		sort.Sort(sort.Reverse(ByInBytes(pairs)))
+	}
+
+	i = 0
```

Comment:
I think we can just redo the slice here and get rid of 'i' and the escape clause, correct?
pairs = pairs[:c.Limit]